Tests/Core/AWSClientRuntimeTests/Retry/AWSRetryErrorInfoProviderTests.swift

```diff
@@ -0,0 +1,135 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+import ClientRuntime
+import AwsCommonRuntimeKit
+@testable import AWSClientRuntime
+
+class AWSRetryErrorInfoProviderTests: XCTestCase {
```

Comment:
Tests for the various cases.

For the cases that match CRT errors, let me know if I'm not matching the right codes or otherwise not handling CRT errors the right way.